### PR TITLE
[rust-sdk]: gracefully close websocket

### PR DESCRIPTION
**Pull request by coolreader18**
# Description of Changes

# API and ABI

 - [ ] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

**Review by gefjon — APPROVED** [review]

**Review by kim — APPROVED** [review]

File: crates/sdk/examples/quickstart-chat/main.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ fn main() {
     connect_to_db();
     subscribe_to_tables();
     user_input_loop();
+    spacetimedb_sdk::global_connection::with_connection_mut(|x| x.disconnect())
```
Comment (gefjon) [inline crates/sdk/examples/quickstart-chat/main.rs:19]: We should expose a global function that does this. `with_connection_mut` is supposed to be an internal detail.

**Review by gefjon — CHANGES_REQUESTED** [review]

File: crates/sdk/src/background_connection.rs [hunk 2]
```diff
@@ -338,12 +336,24 @@ impl BackgroundDbConnection {
...
+    pub fn disconnect(&mut self) {
+        self.send_chan = None;
+        self.client_cache = None;
+        if let Some(h) = self.websocket_loop_handle.take() {
+            let _ = self.handle.block_on(h);
+        }
+        if let Some(h) = self.recv_handle.take() {
+            let _ = self.handle.block_on(h);
+        }
+        self.runtime = None;
```
Comment (gefjon) [inline crates/sdk/src/background_connection.rs:348]: Oh, we also shouldn't be un-setting the runtime - it's required to still exist if we want to re-connect.

**Comment by gefjon** [comment]
Changes requested and made in https://github.com/clockworklabs/SpacetimeDB/pull/234. Once that's in here, we should merge this.

File: crates/sdk/src/background_connection.rs (hunk 2 above)
Comment (coolreader18) [reply crates/sdk/src/background_connection.rs:348]: hmm fair enough

**Review by gefjon — APPROVED** [review]

**Pull request merged**